src/commands/queue.rs

```diff
@@ -48,45 +55,43 @@ pub async fn queue(
             response
                 .kind(InteractionResponseType::ChannelMessageWithSource)
                 .interaction_response_data(|message| {
+                    let num_pages = calculate_num_pages(&tracks);
+
                     message
-                        .create_embed(|e| create_queue_embed(e, &author_username, &tracks, 0))
-                        .components(|components| {
-                            components.create_action_row(|action_row| {
-                                action_row
-                                    .create_button(|button| {
-                                        button
-                                            .custom_id("⏪".to_string().to_ascii_lowercase())
-                                            .emoji(ReactionType::Unicode("⏪".to_string()))
-                                            .style(ButtonStyle::Secondary)
-                                    })
-                                    .create_button(|button| {
-                                        button
-                                            .custom_id("◀️".to_string().to_ascii_lowercase())
-                                            .emoji(ReactionType::Unicode("◀️".to_string()))
-                                            .style(ButtonStyle::Secondary)
-                                    })
-                                    .create_button(|button| {
-                                        button
-                                            .custom_id("▶️".to_string().to_ascii_lowercase())
-                                            .emoji(ReactionType::Unicode("▶️".to_string()))
-                                            .style(ButtonStyle::Secondary)
-                                    })
-                                    .create_button(|button| {
-                                        button
-                                            .custom_id("⏩".to_string().to_ascii_lowercase())
-                                            .emoji(ReactionType::Unicode("⏩".to_string()))
-                                            .style(ButtonStyle::Secondary)
-                                    })
-                            })
-                        })
+                        .add_embed(create_queue_embed(&tracks, 0))
+                        .components(|components| build_nav_btns(components, 0, num_pages))
                 })
         })
         .await?;
 
-    let message = interaction.get_interaction_response(&ctx.http).await?;
+    let mut message = interaction.get_interaction_response(&ctx.http).await?;
+    let page_lock: Arc<RwLock<usize>> = Arc::new(RwLock::new(0));
```

Comment:
In reality, this is the `page`. Only when you call `page.read()` or `page.write()` do you get a lock. In the former a `page_rlock` in the latter a `page_wlock`